### ISSUE TITLE
When using the ML.Dev.Pred.Category.Sig function, it gets stuck at 'There are existing 250 genes in candidate genes, colnames of training data, colnames of validation data'.

### ISSUE THREAD

**Issue by ybl151261** [l-magnificence/Mime#98]
> load("./Example.ici.Rdata")
> res.ici <- ML.Dev.Pred.Category.Sig(train_data = list_train_vali_Data$training,
+                                     list_train_vali_Data = list_train_vali_Data,
+                                     candidate_genes = genelist,
+                                     methods = c('nb','svmRadialWeights','rf','kknn','adaboost','LogitBoost','cancerclass'),
+                                     seed = 5201314,
+                                     cores_for_parallel = 1
+ )
---loading the packages ---
---loading the function---
[1] "There existing 254 genes in candidate genes"
[1] "Intersetion of the candidate genes and the colnames of the provided data"
[1] "There existing 250 genes in candidate genes, colnames of training data, colnames of validation data"
When I run this function, the process gets stuck here without moving for up to ten minutes without any progress. May I ask how much time this step should take to complete, or is it my problem?

**Comment by l-magnificence** [comment]
if cores_for_parallel is 1 and the number of sample in list_train_vali_Data > 1000, thel running time maybe exceed 1 day
